contact/models.py

```diff
@@ -1,6 +1,21 @@
 from django.db import models
+from django.db.models import UniqueConstraint
 
 
 class Information(models.Model):
     title = models.CharField(max_length=255, primary_key=True)
     detail = models.TextField(blank=True)
+
+
+class Chat(models.Model):
+    target_one = models.ForeignKey('target.Target', related_name="chat_one_set", on_delete=models.CASCADE)
+    target_two = models.ForeignKey('target.Target', related_name="chat_two_set", on_delete=models.CASCADE)
+
+    def __str__(self):
+        return f'{self.pk} - {self.target_one.user} {self.target_two.user}'
+
+    class Meta:
+        constraints = [
+            UniqueConstraint(fields=['target_one', 'target_two'], name='unique_chat_between_targets_one_two'),
+            UniqueConstraint(fields=['target_two', 'target_one'], name='unique_chat_between_targets_two_one'),
```

Comment:
Why is this second constraint necesary?
Identity is an equivalence relation, it should partition the sample space

Reply:
Let's say we have`targerA & targetB`. The model allows you to have the Chat object with target A_B but you can also have the Chat object with target B_A however it's the same chat. I tried to make a custom constraint in order to make a unique key `targetA, tarbetB || targetB, targetA` but I couldn't. So I just added those two constraints (you can have the chat A_B and B_A but just one instance of each one).
This situation will not happen in the system flow as when I manage the chats for a new target, I check if there is a chat for A_B or B_A. The idea was to restrict as much as I could the creation of wrong chat objects.

Reply:
cant you use a unique constraint with a [condition](https://docs.djangoproject.com/en/3.0/ref/models/constraints/#condition) to solve this?
You got `and` and `or there`